src/main/kotlin/vm/VM.kt

```diff
@@ -35,30 +33,40 @@ class VM(
     private inline fun popTwo() = listOf(stack.removeFirst(), stack.removeFirst())
     private inline fun popThree() = listOf(stack.removeFirst(), stack.removeFirst(), stack.removeFirst())
     private inline fun popFour() = listOf(stack.removeFirst(), stack.removeFirst(), stack.removeFirst(), stack.removeFirst())
-    private inline fun next() = code[pc++]
+    private inline fun next() = verb.code[pc++]
+
+    fun execute(
+        c: Context,
+        args: List<Value> = listOf(),
+        entryPoint: Int? = null,
+        withVars: Map<String, Value>? = null
```

Comment:
Lambdas excute by passing an entryPoint and withVars, which preloads the variables captured from scope and starts execution at the start of the lambda's code block.